### PR TITLE
"Full build" pipeline for PR's

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [ ] Feature
- [x] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

We have 2 pipeline scripts: one for PR's (test) and one for main (full-build). This means double maintenance and can be harder to debug.

### What is the new behavior?

One full-build pipeline, with conditional steps for the mainline build, mostly signing related.

### Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### Other information

Example non-main build: https://github.com/gaphor/gaphor/actions/runs/2619062283

### CODE REVIEW

File: .github/actions/linux_appimage/action.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: 'Create macOS Application'
```
Comment (danyeaw) [inline .github/actions/linux_appimage/action.yml:1]: The name and description should be updated here

File: .github/actions/macos_dmg/action.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        appstore-connect-username: ${{ inputs.ac_username }}
+        appstore-connect-password: ${{ inputs.ac_password }}
+        primary-bundle-id: org.gaphor.gaphor
+    - name: Staple .dmg
+      if: inputs.mainline_build == 'true'
+      uses: devbotsxyz/xcode-staple@v1.0.0
+      with:
+        product-path: "_packaging/dist/Gaphor-${{ inputs.version }}.dmg"
+    - name: Upload Gaphor-${{ inputs.version }}.dmg
+      if: inputs.mainline_build == 'true'
```
Comment (danyeaw) [inline .github/actions/macos_dmg/action.yml:80]: This was a good idea to only build the dmg on release :+1:

**Review by danyeaw — APPROVED** [review]
This is a great simplification to the build pipeline, I love it @amolenaar!

**Comment by danyeaw** [comment]
@amolenaar It looks like the Linux .tar.gz and *.whl file aren't found and the `${{ steps.setup_and_test.outputs.version }}` is blank.

File: .github/workflows/full-build.yml [hunk 3]
```diff
@@ -145,63 +189,21 @@ jobs:
...
+          base64_encoded_p12: ${{ secrets.BASE64_ENCODED_P12 }}
+          certpassword_p12: ${{ secrets.CERTPASSWORD_P12 }}
+          ac_username:  ${{ secrets.AC_USERNAME }}
+          ac_password: ${{ secrets.AC_PASSWORD }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
 
   check-macos-dmg:
     needs: macos
     runs-on: macos-latest
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/packaging/')
```
Comment (danyeaw) [inline .github/workflows/full-build.yml:206]: Should we just make this `if: ${{ mainline_build == 'true' }}`?

Comment (amolenaar) [reply .github/workflows/full-build.yml:207]: That doesn't work: mainline_build is an env var and therefore can't be used in those job level if statements. Sadly.

**Pull request merged**